pdfgen.c

```diff
@@ -2645,17 +2645,27 @@ static int parse_jpeg_header(struct pdf_img_info *info, const uint8_t *data,
 {
     // See http://www.videotechnology.com/jpeg/j1.html for details
     if (length >= 4 && data[0] == 0xFF && data[1] == 0xD8) {
-        for (size_t i = 0; i < length - 3; i++) {
+        for (size_t i = 2; i < length; i++) {
```

Comment:
Just checking - you're scanning for the jpeg markers, then skipping any leading 0xff bytes inside them, and then using their length marker to know what to skip to check for the next? Why the skipping of 0xff's after the marker, is that part of the jpeg standard?

Reply:
Technically this shouldn't be needed. The standard does not permit empty chunks that just contain a SYNC byte, there is always supposed to be valid segment marker (not 00 or FF) after it. Having said that I had put this into my code a while back because I came across a number of images that seemed to have spurious runs of FF in them, probably created by a dodgy paint program, but it happened more than I expected so I allowed for it.

Reply:
Sorry, I'm not sure, do I click Resolve on this or do you?

Reply:
Yeah, I don't really know how it works either. Resolved :)